activerecord/lib/active_record/relation.rb

```diff
@@ -346,11 +346,14 @@ def compute_cache_version(timestamp_column) # :nodoc:
       if loaded? || distinct_value
         size = records.size
         if size > 0
-          timestamp = max_by(&timestamp_column)._read_attribute(timestamp_column)
+          timestamp = max_by(&timestamp_column).public_send(timestamp_column)
```

Comment:
Using `public_send` will allow the generated method to be overridden in the model, changing how the cache key is calculated when the relation is loaded, which doesn't seem desirable.

Reply:
Right, that was the reason to use `_read_attribute` in the past https://github.com/rails/rails/pull/28776#discussion_r111772279 (we already had used overridden value in `max_by(&timestamp_column)` though).

Anyway, I could change to maintain the purpose of #28776.